nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/serialization/VersionedFlowSynchronizer.java

```diff
@@ -1240,6 +1247,11 @@ public FlowControllerComponentScheduler(final FlowController flowController, fin
 
         @Override
         public void startNow(final Connectable component) {
+            if (ExecutionEngine.STATELESS == component.getProcessGroup().resolveExecutionEngine()) {
+                logger.info("{} should be running but will not start it because its Process Group is configured to run Stateless", component);
```

Comment:
I think it makes sense as an INFO. I don't believe it's a condition that we're really likely to hit, except maybe if an update changes the engine for a Process Group from Standard to Stateless. It that case, I don't know off the top of my head whether this would hit or not. You could argue that maybe WARN makes more sense, if we don't think it's expected. But since there's no real adverse side effect, I think a WARN is stronger than necessary and can cause alarm for no reason.